sdk/storage/storage-blob/MigrationGuide.md

```diff
@@ -0,0 +1,263 @@
+  - [Constructing the clients](#constructing-the-clients)
+  - [Creating a container](#creating-a-container)
+  - [Uploading a blob to the container](#uploading-a-blob-to-the-container)
+  - [Fetching properties of a blob](#fetching-properties-of-a-blob)
+  - [Listing blobs from the container](#listing-blobs-from-the-container)
+  - [Sequential actions](#sequential-actions)
+- [Additional samples](#additional-samples)
+
+## Migration benefits
+
+As Azure has matured and been embraced by a more diverse group of developers, we have been focused on learning the patterns and practices to best support developer productivity and to understand the gaps that the JavaScript client libraries have.
+
+There were several areas of consistent feedback expressed across the Azure client library ecosystem. One of the most important is that the client libraries for different Azure services have not had a consistent approach to organization, naming, and API structure. Additionally, many developers have felt that the learning curve was difficult, and the APIs did not offer a good, approachable, and consistent onboarding story for those learning Azure or exploring a specific Azure service.
+
+To improve the development experience across Azure services, a set of uniform [design guidelines](https://azure.github.io/azure-sdk/general_introduction.html) was created for all languages to drive a consistent experience with established API patterns for all services. A set of [TypeScript & JavaScript Guidelines](https://azure.github.io/azure-sdk/typescript_introduction.html) was also introduced to ensure that TypeScript clients have a natural and idiomatic feel with respect to the TypeScript and JavaScript ecosystems. The new `@azure/storage-blob` client library follows these guidelines.
+
+### Cross Service SDK improvements
+
+The modern `@azure/storage-blob` client library is also benefited from the cross-service improvements made to the Azure development experience, such as
+
+- A unified logging and diagnostics pipeline offering a common view of the activities across each of the client libraries
+- Use of promises rather than callbacks for a simplified programming experience
+- Use of async iterators in paging APIs
+
+## Important changes
+
+### Package name and structure
+
+The modern client library is named `@azure/storage-blob` and was released beginning with version 10. The legacy client library is named `azure-storage` with version of 2.x.x or below.
+
+The legacy library `azure-storage` grouped functionality to work with multiple services in the same package such as `Blob`, `Queue`, `Files` and `Tables`. The new `@azure/storage-blob` is dedicated to `Blob` service. New generation packages are available for the other storage services as well: `@azure/data-tables`, `@azure/storage-queue` and `@azure/storage-file-share`. This provides more granular control on which dependencies to take on your project.
+
+### Constructing the clients
```

Comment:
We are talking about constructing clients using only connection string here. Any reason to not show the differences for other scenarios like using the anonymous credential or AAD?

I would recommend adding those as well. You can have sub headers like
- Constructing clients using connection strings
-  Constructing clients using Azure Active Directory
- etc

Reply:
I added one more sample for blob, queue with AAD credentials and a sample for file with SAS token. 
Samples for other credentials are similar, we may not need to cover all of them.